market/ipni/spark/sol/README.md

```diff
@@ -0,0 +1,302 @@
+# **MinerPeerIDMapping Smart Contract**
+
+## **Overview**
+
+The `MinerPeerIDMapping` contract provides a decentralized way to map Filecoin miner IDs (`uint64`) to their PeerIDs along with signed messages for validation. It ensures only authorized entities (owner or controlling addresses) can add, update, or delete mappings.
+
+---
+
+## **Features**
+
+- **PeerID Management**: Associate PeerIDs with Filecoin miner IDs.
+- **Ownership Control**: Only the owner or authorized controlling addresses can manage mappings.
+- **Signed Message Storage**: Store and verify signed JSON-encoded messages for miner-to-peer mappings.
+- **Ownership Transfers**: Allows the contract owner to transfer ownership to another address.
+
+---
+
+## **Contract Details**
+
+### **State Variables**
+
+- `address public owner`: The contract owner address.
+- `mapping(uint64 => PeerData) public minerToPeerData`: A mapping of miner IDs (`uint64`) to `PeerData`.
+
+---
+
+### **Structs**
+
+- **PeerData**
+  ```solidity
+  struct PeerData {
+      string peerID;
+      bytes signedMessage;
+  }
+  ```
+    - `peerID`: The PeerID associated with the miner.
+    - `signedMessage`: A signed JSON message containing miner and PeerID information.
+
+---
+
+### **Events**
```

Comment:
It's great that you included events! 👏🏻 This allows us to potentially implement a subgraph indexing these events to provide cheaper access than RPC API calls for the current on-chain state.